packages/hurl/src/http/options.rs

```diff
@@ -154,6 +160,17 @@ impl ClientOptions {
             arguments.push("--max-redirs".to_string());
             arguments.push(max_redirect.to_string());
         }
+        if let Some(filename) = &self.netrc_file {
```

Comment:
In this method, I would prefer to make the code "dump",ie it should not have the knowledge of `--netrc`/ `--netrc-optional`being excluse for instance. This will be done in the client.rs file as proposed in this PR, and latter we'll configure Clap options to di this check. For this code for instnec, we can just do:

```rust
        if let Some(filename) = &self.netrc_file {
            arguments.push("--netrc-file".to_string());
            arguments.push(format!("'{filename}'"));
        }
        if self.netrc_optional {
            arguments.push("--netrc-optional".to_string());
        }
        if self.netrc {
            arguments.push("--netrc".to_string());
        }
```

Reviewing the code, I see that we've done this kind ok code for `--location`/`--location-trusted`, I'll change it later.